File: leetcode/python/2942-findWordsContainingCharacter.py
from typing import List


class Solution:
    def findWordsContaining(self, words: List[str], x: str) -> List[int]:
        res = list()
        for i, word in enumerate(words):
            if x in word:
                res.append(i)
        return res


def main():
    words = ["leet","code"]
    x = "e"
    sol = Solution()
    print(sol.findWordsContaining(words, x))


if __name__ == "__main__":
    main()

File: leetcode/go/2942-findWordsContainingCharacter.go
//go:build ignore

package main

import "fmt"

func findWordsContaining(words []string, x byte) []int {
    res := []int{}
	for idx, word := range words {
		for i := range word {
			if word[i] == x {
				res = append(res, idx)
				break
			}
		}
	}
	return res
}

func main() {
	words := []string{"leet", "code"}
	var x byte = 'e'
	fmt.Print(findWordsContaining(words, x))
}